cmake/KokkosConfigCommon.cmake.in

```diff
@@ -14,10 +14,6 @@ ENDFOREACH()
 IF(NOT Kokkos_FIND_QUIETLY)
   MESSAGE(STATUS "Enabled Kokkos devices: ${Kokkos_DEVICES}")
 ENDIF()
-IF(Kokkos_ENABLE_THREADS AND NOT DEPRECATED_CODE_3 IN_LIST Kokkos_OPTIONS)  # for backward compatibility
-  SET(Kokkos_ENABLE_PTHREAD ON)
-  LIST(APPEND Kokkos_DEVICES PTHREAD)
-ENDIF()
```

Comment:
PTHREAD changes needs to be in a separate commit.
And if we are to remove it then certainly we should drop this
https://github.com/kokkos/kokkos/blob/d19aab9981a2c447e832a7b4eb7b16992328fb14/cmake/kokkos_enable_devices.cmake#L23-L29